includes/class-wc-gateway-stripe.php

```diff
@@ -150,7 +150,12 @@ public function is_available() {
 	 * @param string $load_address The address to load.
 	 */
 	public function show_update_card_notice( $user_id, $load_address ) {
-		if ( ! $this->saved_cards || ! WC_Stripe_Payment_Tokens::customer_has_saved_methods( $user_id ) || 'billing' !== $load_address ) {
+		if (
+			is_admin() ||
```

Comment:
This `is_admin()` is the only change within the conditional. The rest is just formatting.

An alternative I contemplated is directly checking whether `wc_add_notice` is defined. I opted  for using `is_admin()` instead because the notice is only intended to be displayed in the frontend, where [the function should always be defined](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-woocommerce.php#L625-L627).

Reply:
I agree, `is_admin()` makes it clear that the notice should only be  shown in the admin dashboard.